documentation/modules/ROOT/pages/configuration/storage.adoc

```diff
@@ -0,0 +1,705 @@
+[id="storing-debezium-state"]
+= Storing state of a {prodname} connector
+ifdef::community[]
+:toc:
+:toc-placement: macro
+:linkattrs:
+:icons: font
+:source-highlighter: highlight.js
+
+toc::[]
+
+== Overview
+endif::community[]
+
+Debezium connectors require a persistent storage to keep their state between restarts.
+All connectors require a store for the offsets.
+In addition to it some of the connectors (MySQL, SQL Server, Db2, Oracle) require an additional storage to record the history of schema evolution, so-called internal schema history.
+
+The offset store is provided automatically for the deployments inside the Kafka Connect runtime:
+
+* Kafka offset store for Kafka Connect distributed
+* File offset store for Kafka Connect standalone
+
+If a connector is run either in xref:development/engine.adoc[Debezium Engine] or xref:operations/debezium-server.adoc[Debezium Server] then the offset store must be explicitly configured.
+The internal schema history is configured in all cases via connector configuration options.
+
+
+== Kafka
+
+Debezium can use Kafka to store its state, including source offsets, schema history. Offsets are stored in a Kafka topic (e.g., `connect-offsets`) via the `KafkaOffsetBackingStore`, ensuring connectors resume from the correct position after restarts. Schema history is saved in a separate compacted topic (e.g., `schema-changes.inventory`).
+
+=== Offset Store
+
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[offset-storage-kafka]]<<offset-storage-kafka,`offset.storage`>>
+|No default
+|Must be set to `org.apache.kafka.connect.storage.KafkaOffsetBackingStore`.
+
+|[[offset-storage-topic]]<<offset-storage-topic,`offset.storage.topic`>>
+|No default
+|Defines the Kafka topic where offsets will be stored. It's essential that this topic is configured with log compaction enabled to retain the latest offset information.
+
+|[[offset-storage-partitions]]<<offset-storage-partitions,`offset.storage.partitions`>>
+|25
+|Indicates the number of partitions for the offset storage topic. This should align with your Kafka cluster's partitioning strategy.
+
+|[[offset-storage-replication-factor]]<<offset-storage-replication-factor,`offset.storage.replication.factor`>>
+|3
+|Sets the replication factor for the offset storage topic, ensuring fault tolerance by replicating data across multiple brokers.
+
+|===
+
+=== Internal schema history store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[schema-history-internal-kafka]]<<schema-history-internal-kafka,`schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.kafka.history.KafkaSchemaHistory`.
+
```

Comment:
If this and similar properties have required values, can they be coded to carry the appropriate defaults?

Reply:
The [same](https://github.com/debezium/debezium/pull/4771/files#r2045071463) as for the `offset` we actually don't have a default value for `schema.history.internal`. Maybe a correct value could be `MemorySchemaHistory`? WDYK @jpechane

Reply:
Again, this is selection to choose the right storage so there can't be sensible default.